app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,598 @@
+        getSupportFragmentManager().findFragmentByTag(RequestEnableLocationServicesDialog.TAG);
+
+    if (prev == null) {
+      RequestEnableLocationServicesDialog dialogFragment =
+          new RequestEnableLocationServicesDialog();
+      dialogFragment.show(fragmentTransaction, RequestEnableLocationServicesDialog.TAG);
+    }
+  }
+
+  private void requestEnableWifiP2pServices() {
+    FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+    Fragment prev =
+        getSupportFragmentManager().findFragmentByTag(RequestEnableWifiP2pServicesDialog.TAG);
+
+    if (prev == null) {
+      RequestEnableWifiP2pServicesDialog dialogFragment = new RequestEnableWifiP2pServicesDialog();
+      dialogFragment.show(fragmentTransaction, RequestEnableWifiP2pServicesDialog.TAG);
+    }
+  }
+
+  @Override
+  protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+
+    switch (requestCode) {
+      case REQUEST_ENABLE_LOCATION_SERVICES: {
+        LocationManager locationManager =
+            (LocationManager) getApplicationContext().getSystemService(Context.LOCATION_SERVICE);
+
+        if (!(locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)
+            || locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER))) {
+          // If neither provider is enabled
+          showToast(this, R.string.permission_refused_location, Toast.LENGTH_LONG);
+        }
+        break;
+      }
+
+      case REQUEST_ENABLE_WIFI_P2P: {
+        if (!wifiDirectManager.isWifiP2pEnabled()) {
+          showToast(this, R.string.discovery_needs_wifi, Toast.LENGTH_LONG);
+        }
+        break;
+      }
+    }
+  }
+
+  /* Miscellaneous helper methods */
+  static void showToast(Context context, int stringResource, int duration) {
+    showToast(context, context.getString(stringResource), duration);
+  }
+
+  static void showToast(Context context, String text, int duration) {
+    Toast.makeText(context, text, duration).show();
+  }
+
+  void cancelAsyncTasks() {
+    if (peerGroupHandshakeAsyncTask != null) {
+      peerGroupHandshakeAsyncTask.cancel(true);
+    }
+
+    if (senderDeviceAsyncTaskArray != null) {
+      for (SenderDeviceAsyncTask task : senderDeviceAsyncTaskArray) {
+        task.cancel(true);
+      }
+    } else if (receiverDeviceAsyncTask != null) {
+      receiverDeviceAsyncTask.cancel(true);
+    }
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    wifiDirectManager.registerWifiDirectBroadcastRecevier();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    wifiDirectManager.unregisterWifiDirectBroadcastRecevier();
+  }
+
+  @Override protected void onDestroy() {
+    super.onDestroy();
+    cancelAsyncTasks();
+  }
+
+  @Override
+  public void onBackPressed() {
+    super.onBackPressed();
+    wifiDirectManager.closeLocalFileTransferActivity();
```

Comment:
you don't need to override onBackPressed, the deault behaviour is to call Activity.finish() which will in turn call onDestroy

Reply:
I removed the `super.onBackPressed()` which finishes the activity, left `closeLocalFileTransferActivity()` as is to take care of disconnecting as of now
[Changes](https://github.com/kiwix/kiwix-android/pull/1276/commits/58c936b4fec2f7589cac2249f324c1d74146667a#diff-e1d59ae6e15dcb502e125fb6e243d15dL576)

Reply:
No no no no, getting rid of a super call is only for disastrous scenarios. It completely breaks inheritance and is one of the most confusing things that can be done. 
Delete this entire `onBackPressed` method, we want back functionality to stay as it is.

What you are doing is creating arbitrary exit points in your code, all you need to care about is that you clean up everything in onDestroy. `onBackPressed` and `finish` both end up in `onDestroy`

Reply:
nononono, removing a super call is resolved for only disastrous situations. Completely breaks inheritance and is very confusing for anyone else who works on it.

Delete this entire method and handle everything in OnDestroy, `onBackPressed` & `finish()` both end up in onDestroy and that is where you tell wifiDirectManager to stop.

Reply:
Sorry about the mess up here, I'll remember to look up the documentation before removing a super.method() call
[Removed the method](https://github.com/kiwix/kiwix-android/pull/1276/commits/c590ebabe9ff6e6880bdb1eb4e1358fc2e5d9c23)

Reply:
Tch stupid double post by myself, githubs comment system is so much more annoying than bitbucket's.